### PR TITLE
Feat/market orders

### PR DESCRIPTION
**Pull request by poly-rodr**
## Types of changes

- [ ] Refactor/enhancement 
- [ ] Bug fix/behavior correction 
- [ ] New feature 
- [ ] Breaking change 
- [ ] Other, additional 

## Status

- [ ] Prefix PR title with `[WIP]` if necessary (changes not yet made).
- [ ] Add tests to cover changes as needed.
- [ ] Update documentation/changelog as needed.
- [ ] Verify all tests run correctly in CI and pass.
- [ ] Ready for review/merge.

### CODE REVIEW

File: py_clob_client/client.py [hunk 1]
```diff
@@ -322,6 +339,44 @@ def create_order(
...
+
+        # add resolve_order_options, or similar
+        tick_size = self.__resolve_tick_size(
+            order_args.token_id,
+            options.tick_size if options else None,
+        )
+        neg_risk = options.neg_risk if options else False
+
+        if order_args.price is None or order_args.price <= 0:
+            p = self.get_price(order_args.token_id, "SELL")
```
Comment (L-Kov) [inline py_clob_client/client.py:359]: so we just use best bid? why always "sell"?

Comment (L-Kov) [reply py_clob_client/client.py:359]: why is price even needed here ig?

Comment (poly-rodr) [reply py_clob_client/client.py:359]: when you use `SELL` as param of the `get_price` function/request, you get the price of the best ask available and that is what we want to match, right? we want to BUY the best ASK available.

In the clob we have:
```go
func (b *BookImpl) GetMarketPrice(side order.Side, tokenId string) (decimal.Decimal, error) {
	if side == order.BUY {
		return b.GetBestBid(tokenId)
	} else {
		return b.GetBestAsk(tokenId)
	}
}
```

Comment (L-Kov) [reply py_clob_client/client.py:359]: gotcha but a market order could be for a sell too right? in which case this wouldn't work. Also this doesn't include depth. I think the TS version of this is flawed too and I think the FE passes a price which is why we don't have issues with it. I think what should happen here is if a price on the market order is not passed we should fetch the relevant book and calculate the weighted price for the size and then use that price to create the market order.

**Pull request merged**